### DISCUSSION TITLE
开个帖子讨论一下，大家觉得memos的核心功能应该包括哪些？

### DISCUSSION THREAD

**Discussion by skyue** [usememos/memos#1252] [General]
第一时间升级到0.11.1。增加了像openai api这样不是特别关键的功能，以及目前并不紧急的界面调整（我个人是喜欢这版界面的）；反而一些核心功能未完善，比如：

1. markdown链接没有解析的bug
2. 截图复制上传到S3服务时，同名导致图片覆盖的问题，这是上个版本的问题，这个版本未修复
3. file system未支持

个人觉得笔记本身的管理应该是最核心的功能，还有很多值得优先完善的，比如：

1. 本地文件方式管理图片/资源。
2. 修复S3上传的文件名bug —— 这个其实也应该基于插件完善，毕竟第三方云存储，不只S3（兼容）
3. 标签的删、改，目前好像也不支持，甚至标签的自定义排序，可能比一些功能都重要。这些都是管理笔记非常重要的一环。
4. 插件系统。—— 不用急着集把各种功能往Core中集成，有了插件系统，与第三方服务交互的功能生态会解决。

毕竟是开源项目，以上只是一家之言，所以发个帖子大家讨论。

**Comment by moulai** [comment]
同意，我也在更新到0.11.1后，主动降级回了0.11.0，主要有以下几个原因：
1. 没有了热力图。
2. 增加了openAI API功能（且不可关闭），然而我并不需要在Memos中使用openAI。
3. 修改为了三栏视图，然而并没有带来实质上的视图优化。一方面，Memos编辑、查看栏目并没有扩大，另一方面，原本可以快速看到捷径和所有标签，现在却必须到最右边去看。

我希望能够完善的地方：
1. 增加一个“返回顶部”按钮。
2. 优化URL识别的解析。目前会出现，把https://之后很长一段文字都识别成超链接的情况。
3. 增加按照“更新时间”对Memos排序的可选设置，并且希望这个设置在主界面（而不是在设置界面）。这个功能在以前的版本中是有的，但是现在好像没有了。
4. 增加Memos整条折叠的设置，即在视图中，如果Memos超过一定字数，就将其折叠一部分。这样能够更快速地浏览所有的Memos。我知道现在可以使用“---”关键字进行折叠，但这个功能并不好用，而且在Memos的手机端会带来显示问题。效果类似这样：
<img width="400" alt="image" src="https://user-images.githubusercontent.com/8275836/222919105-49881dfb-d1db-418a-815e-afd2dcb5c8a4.png">

Agreed, I also downgraded back to 0.11.0 after updating to 0.11.1 for the following reasons:
1. The heatmap feature was removed.
2. OpenAI API functionality was added (and cannot be disabled), but I don't need it in Memos.
3. The view was updated to three-column, but it didn't bring many substantial visual improvements. On one hand, the Memo editing and viewing columns haven't been expanded; on the other hand, shortcuts and all tags can not be reached as quickly as before.

Functions that I hope for improvement:
1. Add a "back-to-top" button.
2. Optimize URL recognition parsing so that long strings of text after "https://" aren't automatically recognized as hyperlinks.
3. Add an optional setting to sort Memos by "update time", and hope that this setting will be available on the main view (rather than in the settings view). This feature was available in previous versions, but seems to have been removed now.
4. Add an option to collapse entire Memos if they exceed a certain length in views so that users can browse through them more quickly without having to use "---" keywords which are not very user-friendly and cause display issues on mobile APP.

**Reply by he-sb** [reply]
>     3. 增加按照“更新时间”对Memos排序的可选设置，并且希望这个设置在主界面（而不是在设置界面）。这个功能在以前的版本中是有的，但是现在好像没有了。

同感，很需要这个

**Comment by moulai** [comment]
> 第一时间升级到0.11.1。增加了像openai api这样不是特别关键的功能，以及目前并不紧急的界面调整（我个人是喜欢这版界面的）；反而一些核心功能未完善，比如：
> 
> 1. markdown链接没有解析的bug
> 2. 截图复制上传到S3服务时，同名导致图片覆盖的问题，这是上个版本的问题，这个版本未修复
> 3. file system未支持
> 
> 个人觉得笔记本身的管理应该是最核心的功能，还有很多值得优先完善的，比如：
> 
> 1. 本地文件方式管理图片/资源。
> 2. 修复S3上传的文件名bug —— 这个其实也应该基于插件完善，毕竟第三方云存储，不只S3（兼容）
> 3. 标签的删、改，目前好像也不支持，甚至标签的自定义排序，可能比一些功能都重要。这些都是管理笔记非常重要的一环。
> 4. 插件系统。—— 不用急着集把各种功能往Core中集成，有了插件系统，与第三方服务交互的功能生态会解决。
> 
> 毕竟是开源项目，以上只是一家之言，所以发个帖子大家讨论。

Translation:
I upgrade Memos to version 0.11.1 as soon as possible. This version update includes non-critical features such as the OpenAI API and interface adjustments that are not currently urgent (personally, I like this version's interface). However, some core functions have not been improved, such as:
- A bug that markdown links are not parsed properly.
- The issue of images being overwritten due to having the same name when uploading to S3 services was a problem in the previous version and has not been fixed in this one.
- File system support is still lacking.

I believe that taking notes should be the most important function, and there are many things for improvement, including:
- Managing images/resources using local file systems.
- Fixing bugs related to uploading files to S3 (which could also be addressed through plugins since third-party cloud storage other than S3 may need compatibility updates).
- Supporting modifying tags. Even custom tag sorting may be more important than some other features as these are all crucial aspects of note management.
- A plugin system would also be beneficial. Rather than rushing to integrate various functions into Core, an ecosystem for interacting with third-party services can be established through plugins.

As an open-source project, these opinions are my own, so let's discuss them together.

**Comment by boojack** [comment]
关于最近更新的三栏布局，其中三个部分的主要功能如下图所示。

![telegram-cloud-photo-size-5-6059737369466222567-y](https://user-images.githubusercontent.com/24653555/222919914-251db346-0ac6-492a-a736-8b958a98fac5.jpg)

目前导航栏里的大多数条目都是以弹窗的形式展示，这种弹窗实现起来比较简单，但是限制较多，比如宽度会比直接展示要小很多。之后也会逐步将每个条目所对应的页面分别放到内容栏中，替换之前的弹窗，这样可以实现一个更美观且好用的单页应用。

**Reply by moulai** [reply]
感谢您的回复！美化视图确实是一个重要的工作，您这样说明以后我就清楚多了，因为刚更新完，觉得去右边找过滤器还不太顺手，之后可能还需要习惯一下。

**Comment by crisqyxw** [comment]
赞成把热力图添加回来，根据Telegram消息，似乎是在重构，期待一下。
(另，OpenAI的API确实感觉不是一个很重要的功能，也许可以默认关闭或者作为插件提供)

**Reply by Zeng1998** [reply]
> 赞成把热力图添加回来，根据Telegram消息，似乎是在重构，期待一下。 (另，OpenAI的API确实感觉不是一个很重要的功能，也许可以默认关闭或者作为插件提供)

对 OpenAI 集成不感兴趣的可以通过自定义 CSS 把按钮去掉就行了，其他都没影响。（我也是这样做的

**Reply by Oaklight** [reply]
后面各家的模型api可能都会有，有的人可能也有自己/公司部署的模型可以用，所以希望openai这个部分可以做一个专栏doc让大家有余力的自己开发

**Comment by leenzhu** [comment]
OpenAI 在国内环境确实不太好用。

建议保留热力图。

**Reply by leenzhu** [reply]
> OpenAI 在国内环境确实不太好用。
> 
> 建议保留热力图。

看到0.11.1更新了，已加入热力图。

**Comment by 52eraja** [comment]
我也喜欢两栏布局 希望可以保留或者多一种页面布局选择

**Comment by muziworld** [comment]
笔记的导入/导出功能

**Reply by akaagiao1** [reply]
同意，多数是小白用户，迁移真的不好搞

**Comment by den13501** [comment]
希望標籤可以支援帶空格的文字

**Comment by eallion** [comment]
Remove Ask AI.

**Comment by theleomac** [comment]
随着memos条数的增多，希望可以加个一键回到顶部的功能

**Comment by redisread** [comment]
标签的重命名我非常需要！

**Comment by ccshao** [comment]
memo内联，类似notion，但没必要太复杂。

**Comment by mebtte** [comment]
评论功能

**Comment by Juskinbo** [comment]
或许支持对memo的批量归档？

**Comment by sprite82y** [comment]
个人觉得，openai这种东西集成在笔记里非常怪，这东西更适合用浏览器插件，或者系统的单独工具。随时唤起输入，或者划词提问。
另外存储问题，只支持S3新的路径方式，我自己部署的minio也用不了，因为是ip访问的。
webdav用的人应该也是比较多的，但是目前好像也没计划。
图片和数据库分离存储也没有做。
一楼说的重名问题，只要把路径和文件名分开就行了，上传的文件名可以改成md5之类的hash

**Comment by F7ionsy** [comment]
想要在记录memos时添加位置信息的功能。

**Comment by lin-snow** [comment]
我期待能保留核心的功能然后增加类似于插件市场的功能,将后续可拓展的功能变成插件来添加

**Comment by towerstreet** [comment]
希望写笔记的时候支持 `html` 等常用的markdown扩展
